bool origin_found
float64 dist_origin
float64 x
float64 y